src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherScheduler.java

```diff
@@ -0,0 +1,81 @@
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/**
+ * @author Theo Pascoli {@literal <theo.pascoli at rte-france.com>}
+ */
+@Component
+@ConditionalOnProperty(name = "scheduler.enable", havingValue = "true")
+public class JobLauncherScheduler {
+    private static final String RUN_BINDING = "run-task";
+    private static final Logger LOGGER = LoggerFactory.getLogger(JobLauncherScheduler.class);
+    private static final String PATTERN_DATE_FORMAT = "yyyy-MM-dd";
+    private static final ZoneId EUROPE_BRUSSELS_ZONE_ID = ZoneId.of("Europe/Brussels");
+
+    private final DateTimeFormatter timestampFormat = DateTimeFormatter.ofPattern(PATTERN_DATE_FORMAT);
+    private final JobLauncherConfigurationProperties jobLauncherConfigurationProperties;
+    private final RestTemplateBuilder restTemplateBuilder;
+    private final StreamBridge streamBridge;
+
+    public JobLauncherScheduler(JobLauncherConfigurationProperties jobLauncherConfigurationProperties, RestTemplateBuilder restTemplateBuilder, StreamBridge streamBridge) {
+        this.jobLauncherConfigurationProperties = jobLauncherConfigurationProperties;
+        this.restTemplateBuilder = restTemplateBuilder;
+        this.streamBridge = streamBridge;
+    }
+
+    @Scheduled(cron = "0 */${scheduler.frequency-in-minutes} ${scheduler.start-hour}-${scheduler.end-hour} * * *")
+    public void automaticTaskStart() {
+        OffsetDateTime startOfDayTimestamp = getstartingTimestamp();
+
+        String requestUrl = jobLauncherConfigurationProperties.getTaskManagerBusinessDateUrl() + timestampFormat.format(startOfDayTimestamp);
+        LOGGER.info("Requesting URL: {}", requestUrl);
+
+        try {
+            ResponseEntity<TaskDto[]> responseEntity = restTemplateBuilder.build().getForEntity(requestUrl, TaskDto[].class);
+
+            if (responseEntity.getBody() != null && responseEntity.getStatusCode() == HttpStatus.OK) {
+                for (TaskDto task : responseEntity.getBody()) {
+
+                    String taskId = task.getId().toString();
+                    // propagate in logs MDC the task id as an extra field to be able to match microservices logs with calculation tasks.
+                    // This should be done only once, as soon as the information to add in mdc is available.
+                    MDC.put("gridcapa-task-id", taskId);
+
+                    if (task.getStatus() == TaskStatus.READY) {
+                        LOGGER.info("Task launched on TS {}", task.getTimestamp());
+                        streamBridge.send(RUN_BINDING, Objects.requireNonNull(task));
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error(String.format("Exception %s", e.getMessage()));
+        }
+    }
+
+    private OffsetDateTime getstartingTimestamp() {
```

Comment:
As we don't need the timestamp , it's better to rename this methode getStartingDate. You could also do the formatting here and return the String